pkg/tenant/organizations.go

```diff
@@ -0,0 +1,74 @@
+package tenant
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	qd "github.com/rotationalio/ensign/pkg/quarterdeck/api/v1"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/middleware"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/tokens"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rs/zerolog/log"
+)
+
+// Organization Detail fetches the details for an organization from Quarterdeck.
+//
+// Route: GET /v1/organizations/:orgID
+func (s *Server) OrganizationDetail(c *gin.Context) {
+	var (
+		ctx    context.Context
+		claims *tokens.Claims
+		err    error
+	)
+
+	// User credentials are required to make the Quarterdeck request
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		log.Error().Err(err).Msg("could not create user context from request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch credentials for authenticated user"))
+		return
+	}
+
+	// User claims are required to verify that the user is in the organization
+	if claims, err = middleware.GetClaims(c); err != nil {
+		log.Error().Err(err).Msg("could not get user claims from context")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch claims for authenticated user"))
+		return
+	}
+
+	// Parse the orgID passed in from the URL
```

Comment:
Do you think this endpoint should just return the user's current organization (and not require the ID param)? That would work just as well for right now but I'm not sure if Beacon will need to be able to detail other organizations that the user is a member of but is not currently selected.